Mlem/App/Globals/Definitions/AppState.swift

```diff
@@ -12,26 +12,56 @@ import Observation
 
 @Observable
 class AppState {
-    private(set) var guestSession: GuestSession!
-    private(set) var activeSessions: [UserSession] = []
+    private(set) var guestSession: GuestSession! {
+        didSet {
+            if oldValue != guestSession {
+                oldValue?.deactivate()
+            }
+        }
+    }
+    
+    private(set) var activeSessions: [UserSession] = [] {
+        didSet {
+            if oldValue != activeSessions {
+                for session in Set(oldValue).subtracting(activeSessions) {
+                    session.deactivate()
+                }
+            }
+        }
+    }
+    
+    /// ``ContentView`` watches this for changes. When it is toggled, the app is refreshed.
+    var appRefreshToggle: Bool = true
     
     private init() {
         self.guestSession = .init(account: AccountsTracker.main.defaultGuestAccount)
-        changeAccount(to: AccountsTracker.main.defaultAccount, deactivateOldGuest: false)
+        setAccount(to: AccountsTracker.main.defaultAccount)
     }
     
-    func changeAccount(to account: any Account) {
-        changeAccount(to: account, deactivateOldGuest: true)
+    /// If `keepPlace` is `nil`, use the value from `UserDefaults`.
+    func changeAccount(to account: any Account, keepPlace: Bool? = nil) {
+        let keepPlace = keepPlace ?? UserDefaults.standard.bool(forKey: "accounts.keepPlace")
```

Comment:
Why `UserDefaults` instead of `AppStorage`?

Reply:
Sure, I can use `@AppStorage` instead 👍